hw/ip/pwrmgr/doc/_index.md

```diff
@@ -0,0 +1,233 @@
+When the handling of the first is completed, it observes there is another pending request and handles accordingly.
+This is done because low power resets and peripheral requested resets lead to different behaviors.
+When the power manager commits to handling a specific request, it sends a reset hint to the reset manager to inform it of what reasons to record should the processor reset.
+
+If for example a low power entry request is seen first by the power manager, it transitions the system to a low power state and informs the reset manager to record “low power exit” as the reset reason.
+If during this transition a watchdog request is observed, the power manager then uses the watchdog as a wakeup cause to resume the system and handle the associated request.
+As this request is handled, the reset manager records a second reset reason of “watchdog”.
+
+Ultimately when control is returned to software, it sees two reset reasons and must handle accordingly.
+
+
+## Wakeup Recording
+
+Similar to [reset handling](#Reset-Request-Handling), wakeup signals are only observed during slow FSM low power; however its recording is continuous until explicitly disabled by software.
+
+The wakeup recording begins when the fast FSM transitions out of ACTIVE state and continues until explicitly disabled by software.
+This ensures wakeup recording is continuous until software has set up the appropriate peripherals to handle future events.
+
+The software is also able to enable recording during ACTIVE state if it so chooses.  The recording enables are just OR’d together for hardware purposes.
+
+
+## Flash Handling
+
+### Power-Up Handling
+
+The AST automatically takes flash out of power down state as part of power manager's power up request.
+Once flash is powered up, the power manager keeps a running count of the flash initialization time.
+When this time has expired, an acknowledgement is forwarded to the flash controller.
+Once the boot ROM is allowed to execute, it is expected to initialize the flash prior to using it.
+This involves 4 steps:
+
+*   Poll flash control register to ensure flash has powered up (this is the same signal mentioned above).
+*   Obtain flash redundancy or configuration information (could be in flash or OTP)
+    *   If in flash, this information is unscrambled
+    *   This item is TBD and must be further discussed.
+*   Program redundancy or configuration information into the flash controller and lock it from further modification
+*   Initialize flash scrambling and programming any excluded regions based on OTP or other information.
+
+There may be additional steps depending on flash design and software needs.
+
+### Power-Down Handling
+
+When the device enters deep sleep, the flash is automatically put into power down mode by the AST.
+When the device exits low power state, the flash is automatically powered up and it is boot ROM responsibility to poll for flash power-up complete.
```

Comment:
This should probably be "boot ROM's responsibility", but I'd probably rewrite as "the responsibility of boot ROM" because it's a bit easier to read.

Reply:
fixed